### PR TITLE
[audiofile/cgltf/libsbml] Update to the latest version

### LINKED ISSUE #19865: [audiofile] update to 1.0.9
Library name: audiofile

New version number: 1.0.9

Other information that may be useful (release notes, etc...)
https://github.com/adamstark/AudioFile/releases/latest

### LINKED ISSUE #19867: [cgltf] update to 1.10
Library name: cgltf

New version number: 1.10

Other information that may be useful (release notes, etc...)
https://github.com/jkuhlmann/cgltf/releases/latest

### LINKED ISSUE #19885: [libsbml] update to 5.19.0
Library name: libsbml

New version number: 5.19.0

Other information that may be useful (release notes, etc...)
https://sourceforge.net/projects/sbml/files/libsbml/5.19.0/stable/

### PR DESCRIPTION
**Pull request by NancyLi1013**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #19865 
  Fixes #19867 
  Fixes #19885

Update `audiofile `to 1.0.9, use cmake to build instead of installing the headers directly. Also export AudioFileConfig.cmake.

Update `cgltf `to 1.10

Update `libsbml `to 5.19.0, update repo from github.

Note:

All features except for `test `for `libsbml ` have passed with the following triplets:

- x86-windows
- x64-windows-static

`test `only support Unix platform.

The usage of `audiofile` has been tested successfully.

### CODE REVIEW

File: ports/audiofile/fix-cmakeLists.patch [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ add_subdirectory (tests)
+-
++endif()
+ #===============================================================================
+-set (CMAKE_SUPPRESS_REGENERATION true)
+\ No newline at end of file
++set (CMAKE_SUPPRESS_REGENERATION true)
++
++#install and export targets
++install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME}Config)
```
Comment (JackBoosY) [inline ports/audiofile/fix-cmakeLists.patch:33]: Can you please test the usage?

File: ports/libsbml/portfile.cmake [hunk 2]
```diff
@@ -1,31 +1,30 @@
...
 
-vcpkg_extract_source_archive_ex(
+vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
-    ARCHIVE ${ARCHIVE}
-    PATCHES fix-linkage-type.patch
+    REPO sbmlteam/libsbml
+    REF 118ffbf11f1a5245cc544c1eac71019d979ecb20 #libSBML-5.19.0
+    SHA512 7fe8b4d594876c6408e01c646187cb1587d0b4e12707a43286150d4e4646841e547bde971de917de1cdfbbb9365172aeac43c8e02f7d354400f9166f0f1c2c3d
+    HEAD_REF stable
```
Comment (JackBoosY) [inline ports/libsbml/portfile.cmake:8]:
```suggestion
    HEAD_REF development
```
The stable branch should always lag behind the development branch.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/audiofile/fix-cmakeLists.patch (hunk 1 above)
Comment (NancyLi1013) [reply ports/audiofile/fix-cmakeLists.patch:33]: Has tested the usage and it can work fine.

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the updates and """""" quotes

**Pull request merged**